marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -316,33 +293,31 @@ private OkHttpClient.Builder newClientBuilder() {
     private void configureSocketFactory(OkHttpClient.Builder clientBuilder, SSLContext sslContext, X509TrustManager trustManager) {
         if (sslContext == null) {
             clientBuilder.socketFactory(new SocketFactoryDelegator(SocketFactory.getDefault()));
-        } else {
-            if (trustManager == null) {
-                // OkHttp requires the trust manager on Java 9 and on Java 8 after 8u251
-                // transitional workaround -- at next backward incompatibility boundary, replace try with thrown exception
-                try {
-                    TrustManagerFactory trustMgrFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-                    trustMgrFactory.init((KeyStore) null);
-                    TrustManager[] trustMgrs = trustMgrFactory.getTrustManagers();
-                    if (trustMgrs == null || trustMgrs.length == 0) {
-                        throw new IllegalArgumentException("no trust manager and could not get default trust manager");
-                    }
-                    if (!(trustMgrs[0] instanceof X509TrustManager)) {
-                        throw new IllegalArgumentException("no trust manager and default is not an X509TrustManager");
-                    }
-                    trustManager = (X509TrustManager) trustMgrs[0];
-                    sslContext.init(null, trustMgrs, null);
-                    clientBuilder.sslSocketFactory(new SSLSocketFactoryDelegator(sslContext.getSocketFactory()), trustManager);
-                } catch (KeyStoreException e) {
-                    throw new IllegalArgumentException("no trust manager and cannot initialize factory for default", e);
-                } catch (NoSuchAlgorithmException e) {
-                    throw new IllegalArgumentException("no trust manager and no algorithm for default manager", e);
-                } catch (KeyManagementException e) {
-                    throw new IllegalArgumentException("no trust manager and cannot initialize context with default", e);
-                }
-            } else {
-                clientBuilder.sslSocketFactory(new SSLSocketFactoryDelegator(sslContext.getSocketFactory()), trustManager);
-            }
+        } else if (trustManager != null) {
```

Comment:
I reworked this a bit to do the "simple" conditions first - life is simple if sslContext is null or if both sslContext and trustManager are not null. Otherwise, things get complicated - we have to build up a default trustManager.